File: src/MessageHandler/MessageHandler.go
package MessageHandler

import (
	"db/MessageHandler/PulsarLib"
)

func SendMessage(topic string, message []byte) {
	client := PulsarLib.InitClient("pulsar://localhost:6650")
	defer (*client).Close()
	producer := PulsarLib.CreateProducer(client, topic)
	defer (*producer).Close()
	PulsarLib.SendMessage(producer, message)
}

func RecieveMessage(topic string, sub_name string) []byte {
	client := PulsarLib.InitClient("pulsar://localhost:6650")
	defer (*client).Close()
	consumer := PulsarLib.CreateConsumer(client, topic, sub_name)
	msg := PulsarLib.ReceiveMessage(consumer)
	PulsarLib.DestroyConsumer(consumer)
	return msg
}

File: src/node-pulsar/MessageHandler/MessageHandler.js
const PulsarLib = require("./PulsarLib/PulsarLib.js");

async function sendMessage(topic, msg){
    // Create a client
    const client = PulsarLib.InitClient("pulsar://localhost:6650");
    // Create a producer
    const producer = await PulsarLib.CreateProducer(client,topic);
    // Send Message
    PulsarLib.SendMessage(producer,msg);
    await producer.flush();
    await producer.close();
    await client.close();
}

async function receiveMessage(topic, sub_name) {
    // Create a client
    const client = PulsarLib.InitClient("pulsar://localhost:6650");
    // Create a consumer
    const consumer = await PulsarLib.CreateConsumer(client,topic,sub_name);
    // Receive messages
    const msg = await PulsarLib.ReceiveMessage(consumer);
    await consumer.close();
    await client.close();
    return msg
}

module.exports = {sendMessage, receiveMessage}